File: Even Fibonacci numbers/problem2.py
def even_fib(n1=0, n2=2, limit=4000000):
	if max(n1, n2) > limit:
		return 0

	n1, n2 = n2, 4*n2+n1

	return even_fib(n1, n2)+n1

print(even_fib())

File: Even Fibonacci numbers/problem2.hs
sumFib :: Integer -> Integer
sumFib limit = sumFib' 0 2 0 limit

sumFib' :: Integer -> Integer -> Integer -> Integer -> Integer
sumFib' a b acc limit
    | b > limit = acc
    | otherwise = sumFib' b (4*b+a) newAcc limit
        where newAcc = b + acc

main :: IO()
main = print(sumFib(4000000)) 